lib/diagnostic.js

```diff
@@ -32,14 +37,25 @@ const any = function any(array, callback) {
 // made up of the array elements for which invoking
 // the callback returns a truthy value
 const select = function select(array, callback) {
-
+let newArray = new Array(0);
+  for (let i = 0; i < array.length; i++)  {
+    if (callback(array[i], i, array) === true)  {
+      newArray.push(array[i]);
+    }
+  }
+  return newArray;
 };
 
 // Question 5
 // Write a function that returns the first element
 // for which invoking the callback returns a truthy value
 const first = function first(array, callback) {
-
+  for (let i = 0; i < array.length; i++)  {
+    if (callback(array[i], i, array) === true)  {
+      return array[i];
+    }
+  }
+  return undefined;
```

Comment:
I would say that you should return `null` rather than `undefined`. You would also get `undefined` if your function didn't return anything at all; choosing `null `makes it clear that you are returning a value (even if that value is 'no results').